### PR TITLE
add a PR template

### PR DESCRIPTION
**Pull request by nden**
This is a PR template with a checklist and a reminder to add a link to the `JP` ticket.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (jdavies-st) [inline .github/pull_request_template.md:1]: `This comment is` or `These comments are`

**Review by jdavies-st — APPROVED** [review]
Horray!

File: .github/pull_request_template.md
Comment (nden) [reply .github/pull_request_template.md:1]: I can't make up my mind ;-)

Comment (hbushouse) [reply .github/pull_request_template.md:1]: "These comments are ..."

File: .github/pull_request_template.md
Comment (hbushouse) [inline .github/pull_request_template.md:12]: This is a minor nit, but many times when we merge a PR it does not technically "close" a JP ticket. It only sets it as resolved and sets the status to "Ready to test". Once someone else has tested and verified it works, then the JP ticket gets closed. So we might want to consider using a different word here, like "Fixes" or "Resolves".

File: .github/pull_request_template.md
Comment (hbushouse) [inline .github/pull_request_template.md:21]: Documentation? You can't be serious!  ;-)

**Review by hbushouse — COMMENTED** [review]
Other things that I often have to fill in or clean up when a PR is created are labels and milestone. Can we add those to the check list too?

**Comment by nden** [comment]
@hbushouse I will add the milestone to the checklist but labels are now created by the bot using the JIRA `affected_software` field (if the ticket starts on JIRA). Or are you talking about other labels?

Of course if the ticket comes from github someone will have to decide whether to push it to JIRA so a maintainer will always have to tweak it.

**Comment by hbushouse** [comment]
I didn't think that the bot did anything with PR's. I thought it only synced up info between github issues and JIRA tickets. And, like I said, I see a lot of empty labels on PRs that I fill in (e.g. look at #5878 right now).

**Comment by nden** [comment]
@hbushouse My bad, you are right. I added `label(s)` to checklist.

**Review by hbushouse — APPROVED** [review]
Theoretically I shouldn't approve this PR until it has a label, milestone, and documentation, but I'll let that slide.  ;-)

File: .github/pull_request_template.md
Comment (jdavies-st) [reply .github/pull_request_template.md]: "All your comment are belong to us."

**Pull request merged**
